lib/index.js

```diff
@@ -74,15 +121,40 @@ function compileBody(str, options){
         return filename;
       },
       read: function (filename) {
+        var str;
+        var plugins = getPlugin('loadRead');
+        for (var i = 0; i < plugins.length; i++) {
+          str = plugins[i](filename, source, options, dependencies);
+          if (str) {
+            debug_sources[filename] = str;
+            return str;
+          }
+        }
         dependencies.push(filename);
-        var str = fs.readFileSync(filename, 'utf8');
+        str = fs.readFileSync(filename, 'utf8');
         debug_sources[filename] = str;
         return str;
       }
     });
+    getPlugin('postLoad').forEach(function (func) {
+      ast = func(ast);
+    });
+
     ast = filters.handleFilters(ast, exports.filters);
```

Comment:
Unrelated to this change, but we should merge `options.filters` with `exports.filters` so that you can do `jade.compile(str, {filters: {'my-custom-filter': function (str) { ... }}})`